# Project:      Price recommander system for Mobile industry

### Use Case: Mobile industry and websites for mobile prices (e.g. whatmobile.com,pk here in Pakistan)

# Data integration, Visualization and ETL
I used 'train.csv' file where i have my dataset for training

In [1]:
import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_3c4c594765474e10870775abb04cbf2f = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='QoPFoBkCOcy5_z9sCxxIlro_taDDVczipnPs_wXceqeF',
    ibm_auth_endpoint="https://iam.ng.bluemix.net/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3-api.us-geo.objectstorage.service.networklayer.com')

body = client_3c4c594765474e10870775abb04cbf2f.get_object(Bucket='defaultikram-donotdelete-pr-ht8dbkrt3n4iv5',Key='train.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

df_data_1 = pd.read_csv(body)
df_data_1.head()

print(type(df_data_1))

Waiting for a Spark session to start...
Spark Initialization Done! ApplicationId = app-20200423192340-0003
KERNEL_ID = 76918be7-d421-43f7-8d02-bcf67bb57a07
<class 'pandas.core.frame.DataFrame'>


In [2]:
from pyspark.sql.types import *


mySchema = StructType([ StructField("battery_power", IntegerType(), True)\

                       ,StructField("blue", IntegerType(), True)\

                       ,StructField("clock_speed", DoubleType(), True)\

                       ,StructField("dual_sim", IntegerType(), True)\

                       ,StructField("fc", IntegerType(), True)\

                       ,StructField("four_g", IntegerType(), True)\

                       ,StructField("int_memory", IntegerType(), True)\

                       ,StructField("m_dep", DoubleType(), True)\

                       ,StructField("mobile_wt", IntegerType(), True)\

                       ,StructField("n_cores", IntegerType(), True)\

                       ,StructField("pc", IntegerType(), True)\

                       ,StructField("px_height", IntegerType(), True)\

                       ,StructField("px_width", IntegerType(), True)\

                       ,StructField("ram", IntegerType(), True)\

                       ,StructField("sc_h", IntegerType(), True)\

                       ,StructField("sc_w", IntegerType(), True)\

                       ,StructField("talk_time", IntegerType(), True)\

                       ,StructField("three_g", IntegerType(), True)\

                       ,StructField("touch_screen", IntegerType(), True)\

                       ,StructField("wifi", IntegerType(), True)\

                       ,StructField("price_range", IntegerType(), True)])

In [3]:

df = spark.createDataFrame(df_data_1,schema=mySchema)
df.show()


+-------------+----+-----------+--------+---+------+----------+-----+---------+-------+---+---------+--------+----+----+----+---------+-------+------------+----+-----------+
|battery_power|blue|clock_speed|dual_sim| fc|four_g|int_memory|m_dep|mobile_wt|n_cores| pc|px_height|px_width| ram|sc_h|sc_w|talk_time|three_g|touch_screen|wifi|price_range|
+-------------+----+-----------+--------+---+------+----------+-----+---------+-------+---+---------+--------+----+----+----+---------+-------+------------+----+-----------+
|          842|   0|        2.2|       0|  1|     0|         7|  0.6|      188|      2|  2|       20|     756|2549|   9|   7|       19|      0|           0|   1|          1|
|         1021|   1|        0.5|       1|  0|     1|        53|  0.7|      136|      3|  6|      905|    1988|2631|  17|   3|        7|      1|           1|   0|          2|
|          563|   1|        0.5|       1|  2|     1|        41|  0.9|      145|      5|  6|     1263|    1716|2603|  11|   2|     

In [4]:
df.printSchema()

root
 |-- battery_power: integer (nullable = true)
 |-- blue: integer (nullable = true)
 |-- clock_speed: double (nullable = true)
 |-- dual_sim: integer (nullable = true)
 |-- fc: integer (nullable = true)
 |-- four_g: integer (nullable = true)
 |-- int_memory: integer (nullable = true)
 |-- m_dep: double (nullable = true)
 |-- mobile_wt: integer (nullable = true)
 |-- n_cores: integer (nullable = true)
 |-- pc: integer (nullable = true)
 |-- px_height: integer (nullable = true)
 |-- px_width: integer (nullable = true)
 |-- ram: integer (nullable = true)
 |-- sc_h: integer (nullable = true)
 |-- sc_w: integer (nullable = true)
 |-- talk_time: integer (nullable = true)
 |-- three_g: integer (nullable = true)
 |-- touch_screen: integer (nullable = true)
 |-- wifi: integer (nullable = true)
 |-- price_range: integer (nullable = true)



In [5]:
df.createOrReplaceTempView('Dataset')

# Initial Data Exploration
Now, I am going to implement some simple functions that i learned earlier so that i can have a view of my dataset. I will find min, mix, standard deviation and some other properties of my data columns and will explore them on different columns. I have imported data using pandas dataframe but now converted it to spark and created a view so that i can run quries on it.

In [6]:
"""
    function to find minimum of a column
"""
def findMin(columnName):
    return spark.sql("SELECT min(battery_power) as minVal from Dataset").first().minVal

In [7]:
"""
    function to find maximum of a column
"""
def findMax(columnName):
    return spark.sql('select max('+columnName+') as maxVal from Dataset').first().maxVal

In [8]:
"""
    function to find mean of a column
"""
def findMean(columnName):
    return spark.sql('select avg('+columnName+') as meanVal from Dataset').first().meanVal

In [9]:
"""
    function to find starndard deviation of a column
"""
def findstdDev(columnName):
    return spark.sql('select stddev('+columnName+') as stddev from Dataset').first().stddev

In [10]:
"""
    function to find skewness for a column
"""
def findSkew(columnName):    
    return spark.sql("""
SELECT 
    (
        1/count(*)
    ) *
    SUM (
        POWER("""+columnName+"""-%s,3)/POWER(%s,3)
    )
    as skewVal
 from Dataset
                    """ %(findMean(columnName),findstdDev(columnName))).first().skewVal


In [11]:
minVal = findMin('battery_power')
print(minVal)

501


In [12]:
findSkew('battery_power')

0.031850640034327575

In [13]:
"""
    function to find kurtosis for specific column
"""
def kurtosis(columnName):    
        return spark.sql("""
SELECT 
    (
        1/count(*)
    ) *
    SUM (
        POWER("""+columnName+"""-%s,4)/POWER(%s,4)
    )
as kut
from Dataset
                    """ %(findMean(columnName),findstdDev(columnName))).first().kut


In [14]:
kurtosis('battery_power')

1.7741403624192582

In [15]:
def correlationinMobileWeightandBatteryPower():
    return spark.sql("SELECT corr(battery_power,mobile_wt) as corrVal from Dataset").first().corrVal

In [16]:
correlationinMobileWeightandBatteryPower()

0.0018443762145708168

# Feature Creation
As i have checked and visualized data above. I would have loved to plot it using matplotlib.pyplot but as data has 21 dimensions so it is impossible to plot hyperdimension here. I have visualized data using different parameters. As i pbtained dataset from Kaggle and checked data. there was no need of big cleaning because data was not redundant or all the columns have values. There is null value in any cell. Now, it is time to create features and define model using sparkml

In [17]:
from pyspark.ml.feature import OneHotEncoder
from pyspark.ml.feature import VectorAssembler
from pyspark.ml.feature import Normalizer


vectorAssembler = VectorAssembler(inputCols=['battery_power','blue','clock_speed','dual_sim', 'fc','four_g',
                                             'int_memory','m_dep','mobile_wt','n_cores', 'pc','px_height','px_width',
                                             'ram','sc_h','sc_w','talk_time','three_g','touch_screen','wifi'], outputCol='features')

normalizer = Normalizer(inputCol='features', outputCol='features_normalized', p=1.0)

# Creating ML Pipeline

In [18]:
from pyspark.ml import Pipeline

pipeline = Pipeline(stages =[vectorAssembler, normalizer])

model = pipeline.fit(df)

new_df = model.transform(df)

new_df.show()

+-------------+----+-----------+--------+---+------+----------+-----+---------+-------+---+---------+--------+----+----+----+---------+-------+------------+----+-----------+--------------------+--------------------+
|battery_power|blue|clock_speed|dual_sim| fc|four_g|int_memory|m_dep|mobile_wt|n_cores| pc|px_height|px_width| ram|sc_h|sc_w|talk_time|three_g|touch_screen|wifi|price_range|            features| features_normalized|
+-------------+----+-----------+--------+---+------+----------+-----+---------+-------+---+---------+--------+----+----+----+---------+-------+------------+----+-----------+--------------------+--------------------+
|          842|   0|        2.2|       0|  1|     0|         7|  0.6|      188|      2|  2|       20|     756|2549|   9|   7|       19|      0|           0|   1|          1|[842.0,0.0,2.2,0....|[0.19111171637387...|
|         1021|   1|        0.5|       1|  0|     1|        53|  0.7|      136|      3|  6|      905|    1988|2631|  17|   3|        7| 

# Deleting useless columns
this now has become a big dataframe and i don't want that. I have created features and normalized. I have performed all of them using ML pipeline availabel in Apache spark. Now its time to delete that columns that i don't need
Let's do it.

In [19]:
df2 = new_df.drop('battery_power').drop('blue').drop('clock_speed').drop('dual_sim').drop('fc')\
.drop('four_g').drop('int_memory').drop('m_dep').drop('mobile_wt').drop('n_cores').drop('pc')\
.drop('px_height').drop('px_width').drop('ram').drop('sc_h').drop('sc_w').drop('talk_time')\
.drop('three_g').drop('touch_screen').drop('wifi')

In [20]:
df2.show()

+-----------+--------------------+--------------------+
|price_range|            features| features_normalized|
+-----------+--------------------+--------------------+
|          1|[842.0,0.0,2.2,0....|[0.19111171637387...|
|          2|[1021.0,1.0,0.5,1...|[0.15067441929104...|
|          2|[563.0,1.0,0.5,1....|[0.08834975833281...|
|          2|[615.0,1.0,2.5,0....|[0.09343238685565...|
|          1|[1821.0,1.0,1.2,0...|[0.30881155881155...|
|          1|[1859.0,0.0,0.5,1...|[0.31984446509067...|
|          3|[1821.0,0.0,1.7,0...|[0.27360829389226...|
|          0|[1954.0,0.0,0.5,1...|[0.42857456188449...|
|          0|[1445.0,1.0,0.5,0...|[0.35633260998224...|
|          0|[509.0,1.0,0.6,1....|[0.14327131477467...|
|          3|[769.0,1.0,2.9,1....|[0.12704444077317...|
|          3|[1520.0,1.0,2.2,0...|[0.22396746577865...|
|          1|[1815.0,0.0,2.8,0...|[0.37105941039375...|
|          2|[803.0,1.0,2.1,0....|[0.14541569330508...|
|          0|[1866.0,0.0,0.5,0...|[0.53989931138

# Machine Learning Algorithms

## 1. Decision Tree Classifier

In [21]:
from pyspark.ml.classification import DecisionTreeClassifier

classifier = DecisionTreeClassifier(featuresCol="features_normalized", labelCol="price_range")

model = classifier.fit(df2)

model = model.transform(df2)

model.show()

+-----------+--------------------+--------------------+--------------------+--------------------+----------+
|price_range|            features| features_normalized|       rawPrediction|         probability|prediction|
+-----------+--------------------+--------------------+--------------------+--------------------+----------+
|          1|[842.0,0.0,2.2,0....|[0.19111171637387...|  [10.0,8.0,0.0,0.0]|[0.55555555555555...|       0.0|
|          2|[1021.0,1.0,0.5,1...|[0.15067441929104...|[0.0,7.0,40.0,365.0]|[0.0,0.0169902912...|       3.0|
|          2|[563.0,1.0,0.5,1....|[0.08834975833281...|[0.0,7.0,40.0,365.0]|[0.0,0.0169902912...|       3.0|
|          2|[615.0,1.0,2.5,0....|[0.09343238685565...|[0.0,7.0,40.0,365.0]|[0.0,0.0169902912...|       3.0|
|          1|[1821.0,1.0,1.2,0...|[0.30881155881155...|[45.0,165.0,40.0,...|[0.18,0.66,0.16,0.0]|       1.0|
|          1|[1859.0,0.0,0.5,1...|[0.31984446509067...|[45.0,165.0,40.0,...|[0.18,0.66,0.16,0.0]|       1.0|
|          3|[1821.

In [22]:
from pyspark.ml.evaluation import MulticlassClassificationEvaluator
binEval = MulticlassClassificationEvaluator().setMetricName("accuracy") .setPredictionCol("prediction").setLabelCol("price_range")
    
binEval.evaluate(model) 

0.8115

In [23]:
df2.show()

+-----------+--------------------+--------------------+
|price_range|            features| features_normalized|
+-----------+--------------------+--------------------+
|          1|[842.0,0.0,2.2,0....|[0.19111171637387...|
|          2|[1021.0,1.0,0.5,1...|[0.15067441929104...|
|          2|[563.0,1.0,0.5,1....|[0.08834975833281...|
|          2|[615.0,1.0,2.5,0....|[0.09343238685565...|
|          1|[1821.0,1.0,1.2,0...|[0.30881155881155...|
|          1|[1859.0,0.0,0.5,1...|[0.31984446509067...|
|          3|[1821.0,0.0,1.7,0...|[0.27360829389226...|
|          0|[1954.0,0.0,0.5,1...|[0.42857456188449...|
|          0|[1445.0,1.0,0.5,0...|[0.35633260998224...|
|          0|[509.0,1.0,0.6,1....|[0.14327131477467...|
|          3|[769.0,1.0,2.9,1....|[0.12704444077317...|
|          3|[1520.0,1.0,2.2,0...|[0.22396746577865...|
|          1|[1815.0,0.0,2.8,0...|[0.37105941039375...|
|          2|[803.0,1.0,2.1,0....|[0.14541569330508...|
|          0|[1866.0,0.0,0.5,0...|[0.53989931138

## 2. Random Forest Classifier

In [24]:
from pyspark.ml.classification import RandomForestClassifier

classifier = RandomForestClassifier(featuresCol="features_normalized", labelCol="price_range")

model1 = classifier.fit(df2)

model1 = model1.transform(df2)

model1.show()

+-----------+--------------------+--------------------+--------------------+--------------------+----------+
|price_range|            features| features_normalized|       rawPrediction|         probability|prediction|
+-----------+--------------------+--------------------+--------------------+--------------------+----------+
|          1|[842.0,0.0,2.2,0....|[0.19111171637387...|[1.98681159578887...|[0.09934057978944...|       2.0|
|          2|[1021.0,1.0,0.5,1...|[0.15067441929104...|[0.96935896239778...|[0.04846794811988...|       3.0|
|          2|[563.0,1.0,0.5,1....|[0.08834975833281...|[0.40987511496948...|[0.02049375574847...|       3.0|
|          2|[615.0,1.0,2.5,0....|[0.09343238685565...|[0.32296565751008...|[0.01614828287550...|       3.0|
|          1|[1821.0,1.0,1.2,0...|[0.30881155881155...|[2.30971701616857...|[0.11548585080842...|       1.0|
|          1|[1859.0,0.0,0.5,1...|[0.31984446509067...|[2.72999534444620...|[0.13649976722231...|       1.0|
|          3|[1821.

In [25]:
from pyspark.ml.evaluation import MulticlassClassificationEvaluator
binEval = MulticlassClassificationEvaluator().setMetricName("accuracy") .setPredictionCol("prediction").setLabelCol("price_range")
    
binEval.evaluate(model1) 

0.8635

## 3. Logistic Regression Classifier

In [26]:
from pyspark.ml.classification import LogisticRegression

classifier =LogisticRegression(featuresCol="features_normalized", labelCol="price_range")

model3 = classifier.fit(df2)

model3 = model3.transform(df2)

model3.show()

+-----------+--------------------+--------------------+--------------------+--------------------+----------+
|price_range|            features| features_normalized|       rawPrediction|         probability|prediction|
+-----------+--------------------+--------------------+--------------------+--------------------+----------+
|          1|[842.0,0.0,2.2,0....|[0.19111171637387...|[-4.6117991971757...|[3.25847702938626...|       2.0|
|          2|[1021.0,1.0,0.5,1...|[0.15067441929104...|[-4.5399931399192...|[7.52007943914062...|       2.0|
|          2|[563.0,1.0,0.5,1....|[0.08834975833281...|[-4.4826035856861...|[7.68929870609152...|       2.0|
|          2|[615.0,1.0,2.5,0....|[0.09343238685565...|[-5.2814555859558...|[2.07710523953717...|       2.0|
|          1|[1821.0,1.0,1.2,0...|[0.30881155881155...|[-1.3744837438914...|[0.01698546786883...|       1.0|
|          1|[1859.0,0.0,0.5,1...|[0.31984446509067...|[0.74947976026659...|[0.08969699656940...|       1.0|
|          3|[1821.

In [27]:
from pyspark.ml.evaluation import MulticlassClassificationEvaluator
binEval = MulticlassClassificationEvaluator().setMetricName("accuracy") .setPredictionCol("prediction").setLabelCol("price_range")
    
binEval.evaluate(model3) 

0.7605

# Accuracy Comparision between ML models

### Decision Tree Classifier ==> 81.15 %
### Random Forest Classifier ==> 85.65 %
### Logistic Regression ==> 76.05 %

by examining above evaluations i have found Random forest best working on this dataset and has more accuracy

# Deep Learning 
I am doing it using pytorch framework

In [28]:
!pip install torch torchvision

  Using cached https://files.pythonhosted.org/packages/13/70/54e9fb010fe1547bc4774716f11ececb81ae5b306c05f090f4461ee13205/torch-1.5.0-cp36-cp36m-manylinux1_x86_64.whl
  Using cached https://files.pythonhosted.org/packages/61/51/aa2770a70f612ce9a2fc7da3a1a93f9ecf8746788256fed6b691f9b31ca9/torchvision-0.6.0-cp36-cp36m-manylinux1_x86_64.whl
  Using cached https://files.pythonhosted.org/packages/3d/fc/4763e5f17ac6e7e7d55f377cde859ca1c5d5ac624441ab45315bc578aa9e/numpy-1.18.3-cp36-cp36m-manylinux1_x86_64.whl
  Using cached https://files.pythonhosted.org/packages/ba/90/8a24e6220cfcf6a3a0162535d5b926e774117e384ff921908e07e4c92bda/Pillow-7.1.1-cp36-cp36m-manylinux1_x86_64.whl
tensorflow 1.13.1 requires tensorboard<1.14.0,>=1.13.0, which is not installed.
Target directory /home/spark/shared/user-libs/python3.6/libfuturize already exists. Specify --upgrade to force replacement.
Target directory /home/spark/shared/user-libs/python3.6/Pillow-7.1.1.dist-info already exists. Specify --upgrade to forc

### Again taking our initial dataset in Pandas dataframe and converting it to pytorch tensor

In [29]:
df_data_1

battery_power  blue  clock_speed  dual_sim  fc  four_g  int_memory  \
0               842     0          2.2         0   1       0           7   
1              1021     1          0.5         1   0       1          53   
2               563     1          0.5         1   2       1          41   
3               615     1          2.5         0   0       0          10   
4              1821     1          1.2         0  13       1          44   
5              1859     0          0.5         1   3       0          22   
6              1821     0          1.7         0   4       1          10   
7              1954     0          0.5         1   0       0          24   
8              1445     1          0.5         0   0       0          53   
9               509     1          0.6         1   2       1           9   
10              769     1          2.9         1   0       0           9   
11             1520     1          2.2         0   5       1          33   
12             1815     0          2.8         0   2       0          33   
13              803     1          2.1         0   7       0          17   
14             1866     0          0.5         0  13       1          52   
15              775     0          1.0         0   3       0          46   
16              838     0          0.5         0   1       1          13   
17              595     0          0.9         1   7       1          23   
18             1131     1          0.5         1  11       0          49   
19              682     1          0.5         0   4       0          19   
20              772     0          1.1         1  12       0          39   
21             1709     1          2.1         0   1       0          13   
22             1949     0          2.6         1   4       0          47   
23             1602     1          2.8         1   4       1          38   
24              503     0          1.2         1   5       1           8   
25              961     1          1.4         1   0       1          57   
26              519     1          1.6         1   7       1          51   
27              956     0          0.5         0   1       1          41   
28             1453     0          1.6         1  12       1          52   
29              851     0          0.5         0   3       0          21   
...             ...   ...          ...       ...  ..     ...         ...   
1970           1913     1          1.8         0   0       0          29   
1971            538     0          1.1         1   0       1          25   
1972           1191     0          0.8         0   6       1          46   
1973            816     0          3.0         1   2       0           9   
1974            915     1          0.5         1   9       1          33   
1975           1157     1          0.8         0   7       0          27   
1976           1201     1          0.5         0   2       0          10   
1977           1379     0          1.1         1   1       1          18   
1978           1483     1          2.2         0   3       1          53   
1979           1614     0          1.2         0   1       1           9   
1980            930     1          1.0         1   4       1           4   
1981           1454     0          2.6         0   8       0           6   
1982           1784     0          1.6         0   4       0          41   
1983           1262     0          1.8         1  12       0          34   
1984            797     0          2.2         1   0       0          37   
1985           1829     1          2.1         0   8       0          59   
1986           1139     1          0.9         1   6       1          58   
1987            618     1          1.0         0   9       1          13   
1988           1547     1          2.9         0   2       0          57   
1989            586     0          2.8         0   2       0          15   
1990           1617     1          2.4         0   8  

In [30]:
import torch
from torch import nn
from torch import optim
import torch.nn.functional as F

In [31]:
label_df = df_data_1['price_range']
df_train_new = df_data_1.drop(labels='price_range', axis=1)

train_data = torch.tensor(df_train_new.values)
print(train_data.shape)

train_label = torch.tensor(label_df.values)
train_label

train_label1 = F.one_hot(train_label, num_classes=4)

torch.Size([2000, 20])


In [32]:
class Classifier(nn.Module):
    
    def __init__(self):
        super().__init__()
        
        self.layer1 = nn.Linear(20,17)
        self.layer2 = nn.Linear(17,15)
        self.layer3 = nn.Linear(15,13)
        self.layer4 = nn.Linear(13,10)
        self.layer5 = nn.Linear(10,8)
        self.layer6 = nn.Linear(8,4)
    
    def forward(self,x):
        
        x = F.relu(self.layer1(x))
        x = F.dropout(x)
        x = F.relu(self.layer2(x))
        x = F.relu(self.layer3(x))
        x = F.dropout(x)
        x = F.relu(self.layer4(x))
        x = F.relu(self.layer5(x)) 
        x = self.layer6(x)
        
        return x
print(train_data.shape)
print(train_label.shape)
print(train_label)

torch.Size([2000, 20])
torch.Size([2000])
tensor([1, 2, 2,  ..., 3, 0, 3])


In [33]:
model = Classifier()

optimizer = optim.Adam(model.parameters(), lr=0.005)

criterian = nn.CrossEntropyLoss()

train_data = train_data.float()
train_lable = train_label.float()

epoch = 10000
for i in range(epoch):
    
    optimizer.zero_grad()
    
    output = model(train_data.float())
    loss = criterian(output,train_label)
    loss.backward()
    optimizer.step()
    
    print('At ',i,'/100 epoch loss is: ',loss.item())

At  0 /100 epoch loss is:  8.767779350280762
At  1 /100 epoch loss is:  5.484195709228516
At  2 /100 epoch loss is:  4.144908905029297
At  3 /100 epoch loss is:  3.1126255989074707
At  4 /100 epoch loss is:  2.597700595855713
At  5 /100 epoch loss is:  2.250540256500244
At  6 /100 epoch loss is:  1.9684960842132568
At  7 /100 epoch loss is:  1.8065251111984253
At  8 /100 epoch loss is:  1.6604270935058594
At  9 /100 epoch loss is:  1.5682059526443481
At  10 /100 epoch loss is:  1.4945271015167236
At  11 /100 epoch loss is:  1.447304606437683
At  12 /100 epoch loss is:  1.4271869659423828
At  13 /100 epoch loss is:  1.418051838874817
At  14 /100 epoch loss is:  1.385413408279419
At  15 /100 epoch loss is:  1.3814194202423096
At  16 /100 epoch loss is:  1.3729850053787231
At  17 /100 epoch loss is:  1.375566005706787
At  18 /100 epoch loss is:  1.362054467201233
At  19 /100 epoch loss is:  1.3701282739639282
At  20 /100 epoch loss is:  1.359167456626892
At  21 /100 epoch loss is:  1.3609

At  180 /100 epoch loss is:  0.9194445013999939
At  181 /100 epoch loss is:  0.9329240322113037
At  182 /100 epoch loss is:  0.9353368878364563
At  183 /100 epoch loss is:  0.9190411567687988
At  184 /100 epoch loss is:  0.9320564866065979
At  185 /100 epoch loss is:  0.9397959113121033
At  186 /100 epoch loss is:  0.9350596070289612
At  187 /100 epoch loss is:  0.9300249814987183
At  188 /100 epoch loss is:  0.9275970458984375
At  189 /100 epoch loss is:  0.9177674055099487
At  190 /100 epoch loss is:  0.9415305256843567
At  191 /100 epoch loss is:  0.9131444692611694
At  192 /100 epoch loss is:  0.9211146831512451
At  193 /100 epoch loss is:  0.9171464443206787
At  194 /100 epoch loss is:  0.910377562046051
At  195 /100 epoch loss is:  0.905182421207428
At  196 /100 epoch loss is:  0.9241195917129517
At  197 /100 epoch loss is:  0.9054490923881531
At  198 /100 epoch loss is:  0.9187336564064026
At  199 /100 epoch loss is:  0.9087516665458679
At  200 /100 epoch loss is:  0.90743994712

At  362 /100 epoch loss is:  0.7815998196601868
At  363 /100 epoch loss is:  0.7863196134567261
At  364 /100 epoch loss is:  0.7743098139762878
At  365 /100 epoch loss is:  0.8081145286560059
At  366 /100 epoch loss is:  0.7718796730041504
At  367 /100 epoch loss is:  0.7763850092887878
At  368 /100 epoch loss is:  0.7901785969734192
At  369 /100 epoch loss is:  0.7600828409194946
At  370 /100 epoch loss is:  0.7721942067146301
At  371 /100 epoch loss is:  0.7667984366416931
At  372 /100 epoch loss is:  0.760758638381958
At  373 /100 epoch loss is:  0.7700400948524475
At  374 /100 epoch loss is:  0.7701669931411743
At  375 /100 epoch loss is:  0.7696515917778015
At  376 /100 epoch loss is:  0.7558673620223999
At  377 /100 epoch loss is:  0.7797437310218811
At  378 /100 epoch loss is:  0.7590242028236389
At  379 /100 epoch loss is:  0.7853897213935852
At  380 /100 epoch loss is:  0.7666328549385071
At  381 /100 epoch loss is:  0.7663819789886475
At  382 /100 epoch loss is:  0.7605540156

At  549 /100 epoch loss is:  0.6209319829940796
At  550 /100 epoch loss is:  0.602697491645813
At  551 /100 epoch loss is:  0.61661297082901
At  552 /100 epoch loss is:  0.6181685924530029
At  553 /100 epoch loss is:  0.614670991897583
At  554 /100 epoch loss is:  0.6074019074440002
At  555 /100 epoch loss is:  0.6175762414932251
At  556 /100 epoch loss is:  0.6127781867980957
At  557 /100 epoch loss is:  0.6463429927825928
At  558 /100 epoch loss is:  0.6155762672424316
At  559 /100 epoch loss is:  0.6189233064651489
At  560 /100 epoch loss is:  0.6075008511543274
At  561 /100 epoch loss is:  0.6307129263877869
At  562 /100 epoch loss is:  0.6177141666412354
At  563 /100 epoch loss is:  0.5980479717254639
At  564 /100 epoch loss is:  0.6012685298919678
At  565 /100 epoch loss is:  0.6141433119773865
At  566 /100 epoch loss is:  0.60199373960495
At  567 /100 epoch loss is:  0.6024884581565857
At  568 /100 epoch loss is:  0.6415778994560242
At  569 /100 epoch loss is:  0.601340293884277

At  724 /100 epoch loss is:  0.4873368442058563
At  725 /100 epoch loss is:  0.4795728921890259
At  726 /100 epoch loss is:  0.5001673102378845
At  727 /100 epoch loss is:  0.4766225218772888
At  728 /100 epoch loss is:  0.4895084798336029
At  729 /100 epoch loss is:  0.473394513130188
At  730 /100 epoch loss is:  0.48530474305152893
At  731 /100 epoch loss is:  0.48739737272262573
At  732 /100 epoch loss is:  0.4749639630317688
At  733 /100 epoch loss is:  0.4868903160095215
At  734 /100 epoch loss is:  0.4622914493083954
At  735 /100 epoch loss is:  0.46829554438591003
At  736 /100 epoch loss is:  0.45399773120880127
At  737 /100 epoch loss is:  0.4769652485847473
At  738 /100 epoch loss is:  0.46388787031173706
At  739 /100 epoch loss is:  0.45649561285972595
At  740 /100 epoch loss is:  0.4797351062297821
At  741 /100 epoch loss is:  0.47855088114738464
At  742 /100 epoch loss is:  0.44810813665390015
At  743 /100 epoch loss is:  0.48720115423202515
At  744 /100 epoch loss is:  0.4

At  906 /100 epoch loss is:  0.43560275435447693
At  907 /100 epoch loss is:  0.41644755005836487
At  908 /100 epoch loss is:  0.4355284869670868
At  909 /100 epoch loss is:  0.42961814999580383
At  910 /100 epoch loss is:  0.43115851283073425
At  911 /100 epoch loss is:  0.4085668921470642
At  912 /100 epoch loss is:  0.4087528884410858
At  913 /100 epoch loss is:  0.40844565629959106
At  914 /100 epoch loss is:  0.42912566661834717
At  915 /100 epoch loss is:  0.39587894082069397
At  916 /100 epoch loss is:  0.4306449592113495
At  917 /100 epoch loss is:  0.4201236665248871
At  918 /100 epoch loss is:  0.41973286867141724
At  919 /100 epoch loss is:  0.4398704767227173
At  920 /100 epoch loss is:  0.3995494544506073
At  921 /100 epoch loss is:  0.4023330807685852
At  922 /100 epoch loss is:  0.39283907413482666
At  923 /100 epoch loss is:  0.40031638741493225
At  924 /100 epoch loss is:  0.4170726537704468
At  925 /100 epoch loss is:  0.39864641427993774
At  926 /100 epoch loss is:  

At  1081 /100 epoch loss is:  0.3545364439487457
At  1082 /100 epoch loss is:  0.36219897866249084
At  1083 /100 epoch loss is:  0.361959844827652
At  1084 /100 epoch loss is:  0.3576965928077698
At  1085 /100 epoch loss is:  0.3681335151195526
At  1086 /100 epoch loss is:  0.36873099207878113
At  1087 /100 epoch loss is:  0.359658420085907
At  1088 /100 epoch loss is:  0.35566338896751404
At  1089 /100 epoch loss is:  0.37266355752944946
At  1090 /100 epoch loss is:  0.374359130859375
At  1091 /100 epoch loss is:  0.3659338057041168
At  1092 /100 epoch loss is:  0.3938581049442291
At  1093 /100 epoch loss is:  0.3630329966545105
At  1094 /100 epoch loss is:  0.3798161745071411
At  1095 /100 epoch loss is:  0.37220853567123413
At  1096 /100 epoch loss is:  0.36015358567237854
At  1097 /100 epoch loss is:  0.37196084856987
At  1098 /100 epoch loss is:  0.3571315109729767
At  1099 /100 epoch loss is:  0.3772801458835602
At  1100 /100 epoch loss is:  0.3466183841228485
At  1101 /100 epoch

At  1256 /100 epoch loss is:  0.3245624601840973
At  1257 /100 epoch loss is:  0.3216271996498108
At  1258 /100 epoch loss is:  0.33678266406059265
At  1259 /100 epoch loss is:  0.32933300733566284
At  1260 /100 epoch loss is:  0.33628416061401367
At  1261 /100 epoch loss is:  0.3264063000679016
At  1262 /100 epoch loss is:  0.3242367208003998
At  1263 /100 epoch loss is:  0.34103918075561523
At  1264 /100 epoch loss is:  0.32578960061073303
At  1265 /100 epoch loss is:  0.32847440242767334
At  1266 /100 epoch loss is:  0.33045780658721924
At  1267 /100 epoch loss is:  0.32314160466194153
At  1268 /100 epoch loss is:  0.31293198466300964
At  1269 /100 epoch loss is:  0.303189218044281
At  1270 /100 epoch loss is:  0.32433316111564636
At  1271 /100 epoch loss is:  0.3230205476284027
At  1272 /100 epoch loss is:  0.32314446568489075
At  1273 /100 epoch loss is:  0.3253766894340515
At  1274 /100 epoch loss is:  0.35453349351882935
At  1275 /100 epoch loss is:  0.3265552818775177
At  1276 

At  1439 /100 epoch loss is:  0.2895905375480652
At  1440 /100 epoch loss is:  0.2939085066318512
At  1441 /100 epoch loss is:  0.3095613420009613
At  1442 /100 epoch loss is:  0.2912590801715851
At  1443 /100 epoch loss is:  0.29035988450050354
At  1444 /100 epoch loss is:  0.3001556694507599
At  1445 /100 epoch loss is:  0.29585394263267517
At  1446 /100 epoch loss is:  0.31129372119903564
At  1447 /100 epoch loss is:  0.3055945932865143
At  1448 /100 epoch loss is:  0.297366738319397
At  1449 /100 epoch loss is:  0.2951381206512451
At  1450 /100 epoch loss is:  0.2978942394256592
At  1451 /100 epoch loss is:  0.2888341248035431
At  1452 /100 epoch loss is:  0.30436432361602783
At  1453 /100 epoch loss is:  0.30637291073799133
At  1454 /100 epoch loss is:  0.3178609013557434
At  1455 /100 epoch loss is:  0.31047478318214417
At  1456 /100 epoch loss is:  0.3057319223880768
At  1457 /100 epoch loss is:  0.3155714273452759
At  1458 /100 epoch loss is:  0.29302310943603516
At  1459 /100 

At  1608 /100 epoch loss is:  0.2767780125141144
At  1609 /100 epoch loss is:  0.2820718288421631
At  1610 /100 epoch loss is:  0.28714728355407715
At  1611 /100 epoch loss is:  0.2932969927787781
At  1612 /100 epoch loss is:  0.2884155213832855
At  1613 /100 epoch loss is:  0.29721638560295105
At  1614 /100 epoch loss is:  0.29312142729759216
At  1615 /100 epoch loss is:  0.3079110085964203
At  1616 /100 epoch loss is:  0.30205634236335754
At  1617 /100 epoch loss is:  0.3016590178012848
At  1618 /100 epoch loss is:  0.28420308232307434
At  1619 /100 epoch loss is:  0.28031155467033386
At  1620 /100 epoch loss is:  0.30959004163742065
At  1621 /100 epoch loss is:  0.3101743161678314
At  1622 /100 epoch loss is:  0.304283082485199
At  1623 /100 epoch loss is:  0.30576419830322266
At  1624 /100 epoch loss is:  0.2936241328716278
At  1625 /100 epoch loss is:  0.2963947057723999
At  1626 /100 epoch loss is:  0.3242954909801483
At  1627 /100 epoch loss is:  0.28478187322616577
At  1628 /10

At  1782 /100 epoch loss is:  0.27944356203079224
At  1783 /100 epoch loss is:  0.2770344018936157
At  1784 /100 epoch loss is:  0.27139243483543396
At  1785 /100 epoch loss is:  0.2775570750236511
At  1786 /100 epoch loss is:  0.2703920602798462
At  1787 /100 epoch loss is:  0.2791196405887604
At  1788 /100 epoch loss is:  0.2735725939273834
At  1789 /100 epoch loss is:  0.28085511922836304
At  1790 /100 epoch loss is:  0.28393736481666565
At  1791 /100 epoch loss is:  0.2669416069984436
At  1792 /100 epoch loss is:  0.2571185529232025
At  1793 /100 epoch loss is:  0.2733946740627289
At  1794 /100 epoch loss is:  0.28794145584106445
At  1795 /100 epoch loss is:  0.2699506878852844
At  1796 /100 epoch loss is:  0.2885266840457916
At  1797 /100 epoch loss is:  0.2645472288131714
At  1798 /100 epoch loss is:  0.27317723631858826
At  1799 /100 epoch loss is:  0.26351359486579895
At  1800 /100 epoch loss is:  0.28182631731033325
At  1801 /100 epoch loss is:  0.29414618015289307
At  1802 /1

At  1948 /100 epoch loss is:  0.26086926460266113
At  1949 /100 epoch loss is:  0.273335725069046
At  1950 /100 epoch loss is:  0.27115970849990845
At  1951 /100 epoch loss is:  0.2768130600452423
At  1952 /100 epoch loss is:  0.2633778154850006
At  1953 /100 epoch loss is:  0.278474360704422
At  1954 /100 epoch loss is:  0.26016122102737427
At  1955 /100 epoch loss is:  0.2572374641895294
At  1956 /100 epoch loss is:  0.24680662155151367
At  1957 /100 epoch loss is:  0.27928173542022705
At  1958 /100 epoch loss is:  0.299037367105484
At  1959 /100 epoch loss is:  0.3122439980506897
At  1960 /100 epoch loss is:  0.29100140929222107
At  1961 /100 epoch loss is:  0.25581657886505127
At  1962 /100 epoch loss is:  0.28831323981285095
At  1963 /100 epoch loss is:  0.274385005235672
At  1964 /100 epoch loss is:  0.2604401707649231
At  1965 /100 epoch loss is:  0.2684822976589203
At  1966 /100 epoch loss is:  0.2651643753051758
At  1967 /100 epoch loss is:  0.26091694831848145
At  1968 /100 e

At  2119 /100 epoch loss is:  0.26700398325920105
At  2120 /100 epoch loss is:  0.2726748287677765
At  2121 /100 epoch loss is:  0.23990429937839508
At  2122 /100 epoch loss is:  0.26280078291893005
At  2123 /100 epoch loss is:  0.2686971426010132
At  2124 /100 epoch loss is:  0.2845667600631714
At  2125 /100 epoch loss is:  0.2752699851989746
At  2126 /100 epoch loss is:  0.2524704337120056
At  2127 /100 epoch loss is:  0.2567537724971771
At  2128 /100 epoch loss is:  0.2857525050640106
At  2129 /100 epoch loss is:  0.27429044246673584
At  2130 /100 epoch loss is:  0.2538776695728302
At  2131 /100 epoch loss is:  0.2551785409450531
At  2132 /100 epoch loss is:  0.2714208662509918
At  2133 /100 epoch loss is:  0.2941150367259979
At  2134 /100 epoch loss is:  0.24564212560653687
At  2135 /100 epoch loss is:  0.2550158202648163
At  2136 /100 epoch loss is:  0.2912609875202179
At  2137 /100 epoch loss is:  0.2640395760536194
At  2138 /100 epoch loss is:  0.2449912577867508
At  2139 /100 e

At  2287 /100 epoch loss is:  0.24460867047309875
At  2288 /100 epoch loss is:  0.25224652886390686
At  2289 /100 epoch loss is:  0.23438617587089539
At  2290 /100 epoch loss is:  0.24253909289836884
At  2291 /100 epoch loss is:  0.25348135828971863
At  2292 /100 epoch loss is:  0.24510380625724792
At  2293 /100 epoch loss is:  0.2399313896894455
At  2294 /100 epoch loss is:  0.2468518614768982
At  2295 /100 epoch loss is:  0.2305418848991394
At  2296 /100 epoch loss is:  0.25260165333747864
At  2297 /100 epoch loss is:  0.25453445315361023
At  2298 /100 epoch loss is:  0.25829580426216125
At  2299 /100 epoch loss is:  0.2388480007648468
At  2300 /100 epoch loss is:  0.2444177269935608
At  2301 /100 epoch loss is:  0.235795259475708
At  2302 /100 epoch loss is:  0.2534797489643097
At  2303 /100 epoch loss is:  0.25537875294685364
At  2304 /100 epoch loss is:  0.23214349150657654
At  2305 /100 epoch loss is:  0.2373349815607071
At  2306 /100 epoch loss is:  0.24301999807357788
At  2307 

At  2456 /100 epoch loss is:  0.22672542929649353
At  2457 /100 epoch loss is:  0.25727379322052
At  2458 /100 epoch loss is:  0.23875552415847778
At  2459 /100 epoch loss is:  0.23852606117725372
At  2460 /100 epoch loss is:  0.2341315746307373
At  2461 /100 epoch loss is:  0.25103628635406494
At  2462 /100 epoch loss is:  0.2519502639770508
At  2463 /100 epoch loss is:  0.2412603348493576
At  2464 /100 epoch loss is:  0.231253981590271
At  2465 /100 epoch loss is:  0.24511775374412537
At  2466 /100 epoch loss is:  0.24483783543109894
At  2467 /100 epoch loss is:  0.2302582859992981
At  2468 /100 epoch loss is:  0.24000437557697296
At  2469 /100 epoch loss is:  0.2516089677810669
At  2470 /100 epoch loss is:  0.24097840487957
At  2471 /100 epoch loss is:  0.2685912847518921
At  2472 /100 epoch loss is:  0.23518875241279602
At  2473 /100 epoch loss is:  0.23668567836284637
At  2474 /100 epoch loss is:  0.2564100921154022
At  2475 /100 epoch loss is:  0.2656674385070801
At  2476 /100 ep

At  2623 /100 epoch loss is:  0.23111611604690552
At  2624 /100 epoch loss is:  0.27751624584198
At  2625 /100 epoch loss is:  0.25605833530426025
At  2626 /100 epoch loss is:  0.26532119512557983
At  2627 /100 epoch loss is:  0.24245919287204742
At  2628 /100 epoch loss is:  0.2534946799278259
At  2629 /100 epoch loss is:  0.2502650022506714
At  2630 /100 epoch loss is:  0.2794536352157593
At  2631 /100 epoch loss is:  0.24771617352962494
At  2632 /100 epoch loss is:  0.23584158718585968
At  2633 /100 epoch loss is:  0.2513335645198822
At  2634 /100 epoch loss is:  0.24179838597774506
At  2635 /100 epoch loss is:  0.24599193036556244
At  2636 /100 epoch loss is:  0.23617210984230042
At  2637 /100 epoch loss is:  0.23283709585666656
At  2638 /100 epoch loss is:  0.23779787123203278
At  2639 /100 epoch loss is:  0.23230674862861633
At  2640 /100 epoch loss is:  0.22517608106136322
At  2641 /100 epoch loss is:  0.24520601332187653
At  2642 /100 epoch loss is:  0.25945451855659485
At  264

At  2790 /100 epoch loss is:  0.22126780450344086
At  2791 /100 epoch loss is:  0.2521998882293701
At  2792 /100 epoch loss is:  0.21250197291374207
At  2793 /100 epoch loss is:  0.23631207644939423
At  2794 /100 epoch loss is:  0.23760803043842316
At  2795 /100 epoch loss is:  0.22671030461788177
At  2796 /100 epoch loss is:  0.22425590455532074
At  2797 /100 epoch loss is:  0.25109922885894775
At  2798 /100 epoch loss is:  0.2349555492401123
At  2799 /100 epoch loss is:  0.23732562363147736
At  2800 /100 epoch loss is:  0.22061075270175934
At  2801 /100 epoch loss is:  0.216732919216156
At  2802 /100 epoch loss is:  0.2305605560541153
At  2803 /100 epoch loss is:  0.23156414926052094
At  2804 /100 epoch loss is:  0.2279357761144638
At  2805 /100 epoch loss is:  0.22464151680469513
At  2806 /100 epoch loss is:  0.21773479878902435
At  2807 /100 epoch loss is:  0.24577954411506653
At  2808 /100 epoch loss is:  0.22827337682247162
At  2809 /100 epoch loss is:  0.21939995884895325
At  28

At  2956 /100 epoch loss is:  0.22671052813529968
At  2957 /100 epoch loss is:  0.24886737763881683
At  2958 /100 epoch loss is:  0.2538158595561981
At  2959 /100 epoch loss is:  0.2352564036846161
At  2960 /100 epoch loss is:  0.22632910311222076
At  2961 /100 epoch loss is:  0.23998615145683289
At  2962 /100 epoch loss is:  0.24163949489593506
At  2963 /100 epoch loss is:  0.21990764141082764
At  2964 /100 epoch loss is:  0.22358496487140656
At  2965 /100 epoch loss is:  0.22731520235538483
At  2966 /100 epoch loss is:  0.2362169474363327
At  2967 /100 epoch loss is:  0.21617980301380157
At  2968 /100 epoch loss is:  0.21211308240890503
At  2969 /100 epoch loss is:  0.22374661266803741
At  2970 /100 epoch loss is:  0.22179560363292694
At  2971 /100 epoch loss is:  0.22392886877059937
At  2972 /100 epoch loss is:  0.21626141667366028
At  2973 /100 epoch loss is:  0.21524889767169952
At  2974 /100 epoch loss is:  0.210934117436409
At  2975 /100 epoch loss is:  0.2059909552335739
At  29

At  3125 /100 epoch loss is:  0.2247961014509201
At  3126 /100 epoch loss is:  0.22657641768455505
At  3127 /100 epoch loss is:  0.23806904256343842
At  3128 /100 epoch loss is:  0.24774695932865143
At  3129 /100 epoch loss is:  0.22574414312839508
At  3130 /100 epoch loss is:  0.23779171705245972
At  3131 /100 epoch loss is:  0.2977071702480316
At  3132 /100 epoch loss is:  0.27647680044174194
At  3133 /100 epoch loss is:  0.2639620006084442
At  3134 /100 epoch loss is:  0.25783827900886536
At  3135 /100 epoch loss is:  0.22913092374801636
At  3136 /100 epoch loss is:  0.2359408438205719
At  3137 /100 epoch loss is:  0.26032382249832153
At  3138 /100 epoch loss is:  0.23032787442207336
At  3139 /100 epoch loss is:  0.24899707734584808
At  3140 /100 epoch loss is:  0.24345837533473969
At  3141 /100 epoch loss is:  0.23311258852481842
At  3142 /100 epoch loss is:  0.23953889310359955
At  3143 /100 epoch loss is:  0.2230711728334427
At  3144 /100 epoch loss is:  0.22520777583122253
At  3

At  3299 /100 epoch loss is:  0.2424878031015396
At  3300 /100 epoch loss is:  0.23326994478702545
At  3301 /100 epoch loss is:  0.2614876627922058
At  3302 /100 epoch loss is:  0.2879393994808197
At  3303 /100 epoch loss is:  0.2669648826122284
At  3304 /100 epoch loss is:  0.2158929407596588
At  3305 /100 epoch loss is:  0.233327716588974
At  3306 /100 epoch loss is:  0.27724161744117737
At  3307 /100 epoch loss is:  0.3310600221157074
At  3308 /100 epoch loss is:  0.2834931015968323
At  3309 /100 epoch loss is:  0.2292453646659851
At  3310 /100 epoch loss is:  0.2303386777639389
At  3311 /100 epoch loss is:  0.26955345273017883
At  3312 /100 epoch loss is:  0.27095019817352295
At  3313 /100 epoch loss is:  0.22548246383666992
At  3314 /100 epoch loss is:  0.2268657684326172
At  3315 /100 epoch loss is:  0.2301826775074005
At  3316 /100 epoch loss is:  0.2259586900472641
At  3317 /100 epoch loss is:  0.24981734156608582
At  3318 /100 epoch loss is:  0.2198910117149353
At  3319 /100 e

At  3470 /100 epoch loss is:  0.19858762621879578
At  3471 /100 epoch loss is:  0.2087845355272293
At  3472 /100 epoch loss is:  0.21927103400230408
At  3473 /100 epoch loss is:  0.20287273824214935
At  3474 /100 epoch loss is:  0.22124461829662323
At  3475 /100 epoch loss is:  0.2062019556760788
At  3476 /100 epoch loss is:  0.21814386546611786
At  3477 /100 epoch loss is:  0.2019646018743515
At  3478 /100 epoch loss is:  0.21051689982414246
At  3479 /100 epoch loss is:  0.20807580649852753
At  3480 /100 epoch loss is:  0.20522834360599518
At  3481 /100 epoch loss is:  0.19153305888175964
At  3482 /100 epoch loss is:  0.19259238243103027
At  3483 /100 epoch loss is:  0.20278286933898926
At  3484 /100 epoch loss is:  0.20193178951740265
At  3485 /100 epoch loss is:  0.2067776769399643
At  3486 /100 epoch loss is:  0.2067418098449707
At  3487 /100 epoch loss is:  0.2052745521068573
At  3488 /100 epoch loss is:  0.1962277740240097
At  3489 /100 epoch loss is:  0.20444418489933014
At  349

At  3648 /100 epoch loss is:  0.20286743342876434
At  3649 /100 epoch loss is:  0.18938840925693512
At  3650 /100 epoch loss is:  0.20088040828704834
At  3651 /100 epoch loss is:  0.21012598276138306
At  3652 /100 epoch loss is:  0.2362273782491684
At  3653 /100 epoch loss is:  0.19840407371520996
At  3654 /100 epoch loss is:  0.2232743203639984
At  3655 /100 epoch loss is:  0.19683177769184113
At  3656 /100 epoch loss is:  0.21862661838531494
At  3657 /100 epoch loss is:  0.2156466245651245
At  3658 /100 epoch loss is:  0.24805063009262085
At  3659 /100 epoch loss is:  0.24753466248512268
At  3660 /100 epoch loss is:  0.23903553187847137
At  3661 /100 epoch loss is:  0.2189912497997284
At  3662 /100 epoch loss is:  0.21861504018306732
At  3663 /100 epoch loss is:  0.20533443987369537
At  3664 /100 epoch loss is:  0.22030918300151825
At  3665 /100 epoch loss is:  0.20914949476718903
At  3666 /100 epoch loss is:  0.21261709928512573
At  3667 /100 epoch loss is:  0.2067042738199234
At  3

At  3823 /100 epoch loss is:  0.21104015409946442
At  3824 /100 epoch loss is:  0.2039117068052292
At  3825 /100 epoch loss is:  0.19606538116931915
At  3826 /100 epoch loss is:  0.21554793417453766
At  3827 /100 epoch loss is:  0.21090613305568695
At  3828 /100 epoch loss is:  0.2084459811449051
At  3829 /100 epoch loss is:  0.2098696529865265
At  3830 /100 epoch loss is:  0.20684531331062317
At  3831 /100 epoch loss is:  0.20689533650875092
At  3832 /100 epoch loss is:  0.212832510471344
At  3833 /100 epoch loss is:  0.19673073291778564
At  3834 /100 epoch loss is:  0.20218075811862946
At  3835 /100 epoch loss is:  0.2052052617073059
At  3836 /100 epoch loss is:  0.20191022753715515
At  3837 /100 epoch loss is:  0.21517843008041382
At  3838 /100 epoch loss is:  0.20465077459812164
At  3839 /100 epoch loss is:  0.20702297985553741
At  3840 /100 epoch loss is:  0.19648639857769012
At  3841 /100 epoch loss is:  0.19394929707050323
At  3842 /100 epoch loss is:  0.19931796193122864
At  38

At  4001 /100 epoch loss is:  0.18792448937892914
At  4002 /100 epoch loss is:  0.2024150788784027
At  4003 /100 epoch loss is:  0.19867147505283356
At  4004 /100 epoch loss is:  0.19541658461093903
At  4005 /100 epoch loss is:  0.1868123710155487
At  4006 /100 epoch loss is:  0.2039966881275177
At  4007 /100 epoch loss is:  0.2056795060634613
At  4008 /100 epoch loss is:  0.1830476075410843
At  4009 /100 epoch loss is:  0.2075439691543579
At  4010 /100 epoch loss is:  0.20390360057353973
At  4011 /100 epoch loss is:  0.19592203199863434
At  4012 /100 epoch loss is:  0.1881149560213089
At  4013 /100 epoch loss is:  0.18960851430892944
At  4014 /100 epoch loss is:  0.21065747737884521
At  4015 /100 epoch loss is:  0.19369396567344666
At  4016 /100 epoch loss is:  0.19318722188472748
At  4017 /100 epoch loss is:  0.2001558542251587
At  4018 /100 epoch loss is:  0.17880301177501678
At  4019 /100 epoch loss is:  0.2180934101343155
At  4020 /100 epoch loss is:  0.2148178666830063
At  4021 /

At  4173 /100 epoch loss is:  0.1934741735458374
At  4174 /100 epoch loss is:  0.1811632215976715
At  4175 /100 epoch loss is:  0.19402967393398285
At  4176 /100 epoch loss is:  0.19622349739074707
At  4177 /100 epoch loss is:  0.1900434046983719
At  4178 /100 epoch loss is:  0.19864974915981293
At  4179 /100 epoch loss is:  0.17786350846290588
At  4180 /100 epoch loss is:  0.19314813613891602
At  4181 /100 epoch loss is:  0.18576176464557648
At  4182 /100 epoch loss is:  0.1958959400653839
At  4183 /100 epoch loss is:  0.19137930870056152
At  4184 /100 epoch loss is:  0.19914816319942474
At  4185 /100 epoch loss is:  0.19555403292179108
At  4186 /100 epoch loss is:  0.18653282523155212
At  4187 /100 epoch loss is:  0.181629478931427
At  4188 /100 epoch loss is:  0.19094987213611603
At  4189 /100 epoch loss is:  0.2191358059644699
At  4190 /100 epoch loss is:  0.19695104658603668
At  4191 /100 epoch loss is:  0.19675564765930176
At  4192 /100 epoch loss is:  0.20032921433448792
At  419

At  4342 /100 epoch loss is:  0.20212781429290771
At  4343 /100 epoch loss is:  0.20815791189670563
At  4344 /100 epoch loss is:  0.1956913322210312
At  4345 /100 epoch loss is:  0.20447790622711182
At  4346 /100 epoch loss is:  0.201445072889328
At  4347 /100 epoch loss is:  0.20170089602470398
At  4348 /100 epoch loss is:  0.2141304910182953
At  4349 /100 epoch loss is:  0.20663075149059296
At  4350 /100 epoch loss is:  0.19422641396522522
At  4351 /100 epoch loss is:  0.1964031606912613
At  4352 /100 epoch loss is:  0.218910351395607
At  4353 /100 epoch loss is:  0.24923309683799744
At  4354 /100 epoch loss is:  0.23520660400390625
At  4355 /100 epoch loss is:  0.18740852177143097
At  4356 /100 epoch loss is:  0.21022619307041168
At  4357 /100 epoch loss is:  0.2616604268550873
At  4358 /100 epoch loss is:  0.2402917444705963
At  4359 /100 epoch loss is:  0.23857879638671875
At  4360 /100 epoch loss is:  0.20141811668872833
At  4361 /100 epoch loss is:  0.18720003962516785
At  4362 

At  4510 /100 epoch loss is:  0.18816381692886353
At  4511 /100 epoch loss is:  0.19980745017528534
At  4512 /100 epoch loss is:  0.18681946396827698
At  4513 /100 epoch loss is:  0.20577560365200043
At  4514 /100 epoch loss is:  0.1980438083410263
At  4515 /100 epoch loss is:  0.19604180753231049
At  4516 /100 epoch loss is:  0.21474330127239227
At  4517 /100 epoch loss is:  0.19677744805812836
At  4518 /100 epoch loss is:  0.18718989193439484
At  4519 /100 epoch loss is:  0.20667590200901031
At  4520 /100 epoch loss is:  0.18596501648426056
At  4521 /100 epoch loss is:  0.1980256587266922
At  4522 /100 epoch loss is:  0.1945219188928604
At  4523 /100 epoch loss is:  0.19418425858020782
At  4524 /100 epoch loss is:  0.20073318481445312
At  4525 /100 epoch loss is:  0.21307942271232605
At  4526 /100 epoch loss is:  0.19471655786037445
At  4527 /100 epoch loss is:  0.20254553854465485
At  4528 /100 epoch loss is:  0.21599601209163666
At  4529 /100 epoch loss is:  0.21250800788402557
At 

At  4677 /100 epoch loss is:  0.20330345630645752
At  4678 /100 epoch loss is:  0.19272682070732117
At  4679 /100 epoch loss is:  0.21446049213409424
At  4680 /100 epoch loss is:  0.1779056042432785
At  4681 /100 epoch loss is:  0.20105768740177155
At  4682 /100 epoch loss is:  0.19222021102905273
At  4683 /100 epoch loss is:  0.20410582423210144
At  4684 /100 epoch loss is:  0.20329973101615906
At  4685 /100 epoch loss is:  0.19223624467849731
At  4686 /100 epoch loss is:  0.2037380039691925
At  4687 /100 epoch loss is:  0.21834111213684082
At  4688 /100 epoch loss is:  0.19960543513298035
At  4689 /100 epoch loss is:  0.2066141813993454
At  4690 /100 epoch loss is:  0.18531660735607147
At  4691 /100 epoch loss is:  0.1938612163066864
At  4692 /100 epoch loss is:  0.22155596315860748
At  4693 /100 epoch loss is:  0.19801199436187744
At  4694 /100 epoch loss is:  0.1974785476922989
At  4695 /100 epoch loss is:  0.1883527636528015
At  4696 /100 epoch loss is:  0.19286635518074036
At  46

At  4848 /100 epoch loss is:  0.2012627273797989
At  4849 /100 epoch loss is:  0.1921231746673584
At  4850 /100 epoch loss is:  0.18024684488773346
At  4851 /100 epoch loss is:  0.16898486018180847
At  4852 /100 epoch loss is:  0.18987153470516205
At  4853 /100 epoch loss is:  0.17661947011947632
At  4854 /100 epoch loss is:  0.19347558915615082
At  4855 /100 epoch loss is:  0.18447206914424896
At  4856 /100 epoch loss is:  0.1886637806892395
At  4857 /100 epoch loss is:  0.2404567450284958
At  4858 /100 epoch loss is:  0.23470470309257507
At  4859 /100 epoch loss is:  0.2174689769744873
At  4860 /100 epoch loss is:  0.2036810666322708
At  4861 /100 epoch loss is:  0.20650355517864227
At  4862 /100 epoch loss is:  0.1987501084804535
At  4863 /100 epoch loss is:  0.19118563830852509
At  4864 /100 epoch loss is:  0.21900928020477295
At  4865 /100 epoch loss is:  0.19742773473262787
At  4866 /100 epoch loss is:  0.18959079682826996
At  4867 /100 epoch loss is:  0.17917859554290771
At  486

At  5019 /100 epoch loss is:  0.2361210286617279
At  5020 /100 epoch loss is:  0.19750440120697021
At  5021 /100 epoch loss is:  0.19303958117961884
At  5022 /100 epoch loss is:  0.19668824970722198
At  5023 /100 epoch loss is:  0.187546044588089
At  5024 /100 epoch loss is:  0.1876426488161087
At  5025 /100 epoch loss is:  0.20200495421886444
At  5026 /100 epoch loss is:  0.1993524581193924
At  5027 /100 epoch loss is:  0.19969770312309265
At  5028 /100 epoch loss is:  0.1943402737379074
At  5029 /100 epoch loss is:  0.1947866529226303
At  5030 /100 epoch loss is:  0.2120204120874405
At  5031 /100 epoch loss is:  0.23929482698440552
At  5032 /100 epoch loss is:  0.19258107244968414
At  5033 /100 epoch loss is:  0.19232723116874695
At  5034 /100 epoch loss is:  0.19783030450344086
At  5035 /100 epoch loss is:  0.20314987003803253
At  5036 /100 epoch loss is:  0.19678859412670135
At  5037 /100 epoch loss is:  0.2071419507265091
At  5038 /100 epoch loss is:  0.20454785227775574
At  5039 

At  5195 /100 epoch loss is:  0.20778129994869232
At  5196 /100 epoch loss is:  0.19772806763648987
At  5197 /100 epoch loss is:  0.19502507150173187
At  5198 /100 epoch loss is:  0.19871863722801208
At  5199 /100 epoch loss is:  0.18755485117435455
At  5200 /100 epoch loss is:  0.18074163794517517
At  5201 /100 epoch loss is:  0.18801777064800262
At  5202 /100 epoch loss is:  0.19887106120586395
At  5203 /100 epoch loss is:  0.1848905235528946
At  5204 /100 epoch loss is:  0.17827975749969482
At  5205 /100 epoch loss is:  0.1904318630695343
At  5206 /100 epoch loss is:  0.18696315586566925
At  5207 /100 epoch loss is:  0.18408717215061188
At  5208 /100 epoch loss is:  0.1853494793176651
At  5209 /100 epoch loss is:  0.17184579372406006
At  5210 /100 epoch loss is:  0.19720707833766937
At  5211 /100 epoch loss is:  0.1880102902650833
At  5212 /100 epoch loss is:  0.17373034358024597
At  5213 /100 epoch loss is:  0.18909777700901031
At  5214 /100 epoch loss is:  0.17558354139328003
At  

At  5369 /100 epoch loss is:  0.17974238097667694
At  5370 /100 epoch loss is:  0.17231722176074982
At  5371 /100 epoch loss is:  0.18344652652740479
At  5372 /100 epoch loss is:  0.1757459044456482
At  5373 /100 epoch loss is:  0.17087769508361816
At  5374 /100 epoch loss is:  0.19087101519107819
At  5375 /100 epoch loss is:  0.18888284265995026
At  5376 /100 epoch loss is:  0.19269327819347382
At  5377 /100 epoch loss is:  0.17314070463180542
At  5378 /100 epoch loss is:  0.18830107152462006
At  5379 /100 epoch loss is:  0.1837928742170334
At  5380 /100 epoch loss is:  0.1892077773809433
At  5381 /100 epoch loss is:  0.2017960399389267
At  5382 /100 epoch loss is:  0.22309190034866333
At  5383 /100 epoch loss is:  0.20043018460273743
At  5384 /100 epoch loss is:  0.19199120998382568
At  5385 /100 epoch loss is:  0.19089141488075256
At  5386 /100 epoch loss is:  0.21389493346214294
At  5387 /100 epoch loss is:  0.2184104025363922
At  5388 /100 epoch loss is:  0.1959429532289505
At  53

At  5549 /100 epoch loss is:  0.19123248755931854
At  5550 /100 epoch loss is:  0.1953301876783371
At  5551 /100 epoch loss is:  0.19711469113826752
At  5552 /100 epoch loss is:  0.18506111204624176
At  5553 /100 epoch loss is:  0.20791424810886383
At  5554 /100 epoch loss is:  0.2011488378047943
At  5555 /100 epoch loss is:  0.1976305991411209
At  5556 /100 epoch loss is:  0.1980166733264923
At  5557 /100 epoch loss is:  0.19269277155399323
At  5558 /100 epoch loss is:  0.18928375840187073
At  5559 /100 epoch loss is:  0.19816388189792633
At  5560 /100 epoch loss is:  0.1652074158191681
At  5561 /100 epoch loss is:  0.2057652324438095
At  5562 /100 epoch loss is:  0.2000913918018341
At  5563 /100 epoch loss is:  0.19040420651435852
At  5564 /100 epoch loss is:  0.21348556876182556
At  5565 /100 epoch loss is:  0.21238435804843903
At  5566 /100 epoch loss is:  0.2093016654253006
At  5567 /100 epoch loss is:  0.228537917137146
At  5568 /100 epoch loss is:  0.20090681314468384
At  5569 /

At  5728 /100 epoch loss is:  0.1836317479610443
At  5729 /100 epoch loss is:  0.20527763664722443
At  5730 /100 epoch loss is:  0.20252583920955658
At  5731 /100 epoch loss is:  0.18596968054771423
At  5732 /100 epoch loss is:  0.1830899566411972
At  5733 /100 epoch loss is:  0.1956990361213684
At  5734 /100 epoch loss is:  0.21214602887630463
At  5735 /100 epoch loss is:  0.23458629846572876
At  5736 /100 epoch loss is:  0.19457468390464783
At  5737 /100 epoch loss is:  0.1964544802904129
At  5738 /100 epoch loss is:  0.21898594498634338
At  5739 /100 epoch loss is:  0.2998088300228119
At  5740 /100 epoch loss is:  0.31858035922050476
At  5741 /100 epoch loss is:  0.26805195212364197
At  5742 /100 epoch loss is:  0.20314310491085052
At  5743 /100 epoch loss is:  0.21296478807926178
At  5744 /100 epoch loss is:  0.2812894284725189
At  5745 /100 epoch loss is:  0.28276851773262024
At  5746 /100 epoch loss is:  0.2134649157524109
At  5747 /100 epoch loss is:  0.18987520039081573
At  574

At  5894 /100 epoch loss is:  0.18427373468875885
At  5895 /100 epoch loss is:  0.19086310267448425
At  5896 /100 epoch loss is:  0.18501600623130798
At  5897 /100 epoch loss is:  0.19584600627422333
At  5898 /100 epoch loss is:  0.17125187814235687
At  5899 /100 epoch loss is:  0.19081643223762512
At  5900 /100 epoch loss is:  0.1762344241142273
At  5901 /100 epoch loss is:  0.18066641688346863
At  5902 /100 epoch loss is:  0.176896870136261
At  5903 /100 epoch loss is:  0.19613045454025269
At  5904 /100 epoch loss is:  0.16410717368125916
At  5905 /100 epoch loss is:  0.17400962114334106
At  5906 /100 epoch loss is:  0.16644461452960968
At  5907 /100 epoch loss is:  0.17584335803985596
At  5908 /100 epoch loss is:  0.18458187580108643
At  5909 /100 epoch loss is:  0.18262925744056702
At  5910 /100 epoch loss is:  0.1680743396282196
At  5911 /100 epoch loss is:  0.18568289279937744
At  5912 /100 epoch loss is:  0.17225953936576843
At  5913 /100 epoch loss is:  0.1818135380744934
At  5

At  6062 /100 epoch loss is:  0.189356729388237
At  6063 /100 epoch loss is:  0.18740080296993256
At  6064 /100 epoch loss is:  0.19216787815093994
At  6065 /100 epoch loss is:  0.1808924376964569
At  6066 /100 epoch loss is:  0.16351716220378876
At  6067 /100 epoch loss is:  0.18193387985229492
At  6068 /100 epoch loss is:  0.19002225995063782
At  6069 /100 epoch loss is:  0.18249006569385529
At  6070 /100 epoch loss is:  0.17527931928634644
At  6071 /100 epoch loss is:  0.1665337085723877
At  6072 /100 epoch loss is:  0.1844959706068039
At  6073 /100 epoch loss is:  0.1797887533903122
At  6074 /100 epoch loss is:  0.18349231779575348
At  6075 /100 epoch loss is:  0.16652508080005646
At  6076 /100 epoch loss is:  0.1771959364414215
At  6077 /100 epoch loss is:  0.169357031583786
At  6078 /100 epoch loss is:  0.18753597140312195
At  6079 /100 epoch loss is:  0.18033604323863983
At  6080 /100 epoch loss is:  0.17503869533538818
At  6081 /100 epoch loss is:  0.18625646829605103
At  6082 

At  6237 /100 epoch loss is:  0.19311660528182983
At  6238 /100 epoch loss is:  0.17999424040317535
At  6239 /100 epoch loss is:  0.18373489379882812
At  6240 /100 epoch loss is:  0.17331431806087494
At  6241 /100 epoch loss is:  0.17727035284042358
At  6242 /100 epoch loss is:  0.16780181229114532
At  6243 /100 epoch loss is:  0.18368574976921082
At  6244 /100 epoch loss is:  0.20803141593933105
At  6245 /100 epoch loss is:  0.185554638504982
At  6246 /100 epoch loss is:  0.20272879302501678
At  6247 /100 epoch loss is:  0.19758543372154236
At  6248 /100 epoch loss is:  0.15486247837543488
At  6249 /100 epoch loss is:  0.19742152094841003
At  6250 /100 epoch loss is:  0.210056334733963
At  6251 /100 epoch loss is:  0.18922476470470428
At  6252 /100 epoch loss is:  0.16657514870166779
At  6253 /100 epoch loss is:  0.19513356685638428
At  6254 /100 epoch loss is:  0.1807873547077179
At  6255 /100 epoch loss is:  0.1735701560974121
At  6256 /100 epoch loss is:  0.1633705049753189
At  625

At  6406 /100 epoch loss is:  0.17142324149608612
At  6407 /100 epoch loss is:  0.19576144218444824
At  6408 /100 epoch loss is:  0.1976572573184967
At  6409 /100 epoch loss is:  0.17979420721530914
At  6410 /100 epoch loss is:  0.1969204843044281
At  6411 /100 epoch loss is:  0.1934419870376587
At  6412 /100 epoch loss is:  0.18794739246368408
At  6413 /100 epoch loss is:  0.1997397243976593
At  6414 /100 epoch loss is:  0.17602360248565674
At  6415 /100 epoch loss is:  0.18308019638061523
At  6416 /100 epoch loss is:  0.19833089411258698
At  6417 /100 epoch loss is:  0.1892641931772232
At  6418 /100 epoch loss is:  0.1826808899641037
At  6419 /100 epoch loss is:  0.17582884430885315
At  6420 /100 epoch loss is:  0.1892973929643631
At  6421 /100 epoch loss is:  0.18091517686843872
At  6422 /100 epoch loss is:  0.18440017104148865
At  6423 /100 epoch loss is:  0.1944742500782013
At  6424 /100 epoch loss is:  0.18363994359970093
At  6425 /100 epoch loss is:  0.18291603028774261
At  6426

At  6581 /100 epoch loss is:  0.19092924892902374
At  6582 /100 epoch loss is:  0.1773843616247177
At  6583 /100 epoch loss is:  0.20358185470104218
At  6584 /100 epoch loss is:  0.19495347142219543
At  6585 /100 epoch loss is:  0.23445308208465576
At  6586 /100 epoch loss is:  0.1797335296869278
At  6587 /100 epoch loss is:  0.19296076893806458
At  6588 /100 epoch loss is:  0.2334255874156952
At  6589 /100 epoch loss is:  0.26745933294296265
At  6590 /100 epoch loss is:  0.23801729083061218
At  6591 /100 epoch loss is:  0.19680508971214294
At  6592 /100 epoch loss is:  0.2799026370048523
At  6593 /100 epoch loss is:  0.25502732396125793
At  6594 /100 epoch loss is:  0.20448742806911469
At  6595 /100 epoch loss is:  0.20728032290935516
At  6596 /100 epoch loss is:  0.25086379051208496
At  6597 /100 epoch loss is:  0.24325031042099
At  6598 /100 epoch loss is:  0.19120414555072784
At  6599 /100 epoch loss is:  0.23814649879932404
At  6600 /100 epoch loss is:  0.24975383281707764
At  660

At  6751 /100 epoch loss is:  0.1752289980649948
At  6752 /100 epoch loss is:  0.16561393439769745
At  6753 /100 epoch loss is:  0.1894066035747528
At  6754 /100 epoch loss is:  0.1905161291360855
At  6755 /100 epoch loss is:  0.1897658109664917
At  6756 /100 epoch loss is:  0.17554011940956116
At  6757 /100 epoch loss is:  0.1800236850976944
At  6758 /100 epoch loss is:  0.19562272727489471
At  6759 /100 epoch loss is:  0.18138724565505981
At  6760 /100 epoch loss is:  0.17034479975700378
At  6761 /100 epoch loss is:  0.17118322849273682
At  6762 /100 epoch loss is:  0.17093002796173096
At  6763 /100 epoch loss is:  0.16578100621700287
At  6764 /100 epoch loss is:  0.18127095699310303
At  6765 /100 epoch loss is:  0.16612324118614197
At  6766 /100 epoch loss is:  0.17698347568511963
At  6767 /100 epoch loss is:  0.18093714118003845
At  6768 /100 epoch loss is:  0.19608835875988007
At  6769 /100 epoch loss is:  0.17279097437858582
At  6770 /100 epoch loss is:  0.1903025507926941
At  67

At  6925 /100 epoch loss is:  0.1924700289964676
At  6926 /100 epoch loss is:  0.20966888964176178
At  6927 /100 epoch loss is:  0.23270118236541748
At  6928 /100 epoch loss is:  0.2018778920173645
At  6929 /100 epoch loss is:  0.202257439494133
At  6930 /100 epoch loss is:  0.20162947475910187
At  6931 /100 epoch loss is:  0.19125114381313324
At  6932 /100 epoch loss is:  0.18164141476154327
At  6933 /100 epoch loss is:  0.19010856747627258
At  6934 /100 epoch loss is:  0.20308060944080353
At  6935 /100 epoch loss is:  0.16185887157917023
At  6936 /100 epoch loss is:  0.18315471708774567
At  6937 /100 epoch loss is:  0.17920884490013123
At  6938 /100 epoch loss is:  0.17641125619411469
At  6939 /100 epoch loss is:  0.15920688211917877
At  6940 /100 epoch loss is:  0.16835102438926697
At  6941 /100 epoch loss is:  0.17783993482589722
At  6942 /100 epoch loss is:  0.16947083175182343
At  6943 /100 epoch loss is:  0.18081972002983093
At  6944 /100 epoch loss is:  0.17851434648036957
At  

At  7093 /100 epoch loss is:  0.19617410004138947
At  7094 /100 epoch loss is:  0.16475550830364227
At  7095 /100 epoch loss is:  0.17716138064861298
At  7096 /100 epoch loss is:  0.18219827115535736
At  7097 /100 epoch loss is:  0.1810920238494873
At  7098 /100 epoch loss is:  0.1633751094341278
At  7099 /100 epoch loss is:  0.18077147006988525
At  7100 /100 epoch loss is:  0.1733485460281372
At  7101 /100 epoch loss is:  0.17987652122974396
At  7102 /100 epoch loss is:  0.17909610271453857
At  7103 /100 epoch loss is:  0.17348062992095947
At  7104 /100 epoch loss is:  0.1983635574579239
At  7105 /100 epoch loss is:  0.1744993031024933
At  7106 /100 epoch loss is:  0.17823530733585358
At  7107 /100 epoch loss is:  0.1728539764881134
At  7108 /100 epoch loss is:  0.16490472853183746
At  7109 /100 epoch loss is:  0.18282045423984528
At  7110 /100 epoch loss is:  0.17821285128593445
At  7111 /100 epoch loss is:  0.1673211306333542
At  7112 /100 epoch loss is:  0.17340487241744995
At  711

At  7259 /100 epoch loss is:  0.17504926025867462
At  7260 /100 epoch loss is:  0.16551385819911957
At  7261 /100 epoch loss is:  0.17085643112659454
At  7262 /100 epoch loss is:  0.16114649176597595
At  7263 /100 epoch loss is:  0.1863502413034439
At  7264 /100 epoch loss is:  0.1774454265832901
At  7265 /100 epoch loss is:  0.19099631905555725
At  7266 /100 epoch loss is:  0.17687614262104034
At  7267 /100 epoch loss is:  0.18291215598583221
At  7268 /100 epoch loss is:  0.1783086508512497
At  7269 /100 epoch loss is:  0.1808183640241623
At  7270 /100 epoch loss is:  0.17320910096168518
At  7271 /100 epoch loss is:  0.18444783985614777
At  7272 /100 epoch loss is:  0.1798570156097412
At  7273 /100 epoch loss is:  0.17924374341964722
At  7274 /100 epoch loss is:  0.15939635038375854
At  7275 /100 epoch loss is:  0.1917855441570282
At  7276 /100 epoch loss is:  0.1889302283525467
At  7277 /100 epoch loss is:  0.1877625435590744
At  7278 /100 epoch loss is:  0.18069468438625336
At  7279

At  7430 /100 epoch loss is:  0.16292521357536316
At  7431 /100 epoch loss is:  0.1643284112215042
At  7432 /100 epoch loss is:  0.17706246674060822
At  7433 /100 epoch loss is:  0.17204886674880981
At  7434 /100 epoch loss is:  0.17973102629184723
At  7435 /100 epoch loss is:  0.17217102646827698
At  7436 /100 epoch loss is:  0.17066457867622375
At  7437 /100 epoch loss is:  0.15541666746139526
At  7438 /100 epoch loss is:  0.1627904772758484
At  7439 /100 epoch loss is:  0.16589123010635376
At  7440 /100 epoch loss is:  0.16735303401947021
At  7441 /100 epoch loss is:  0.15652485191822052
At  7442 /100 epoch loss is:  0.18197491765022278
At  7443 /100 epoch loss is:  0.1410052478313446
At  7444 /100 epoch loss is:  0.18391133844852448
At  7445 /100 epoch loss is:  0.16197676956653595
At  7446 /100 epoch loss is:  0.1904352605342865
At  7447 /100 epoch loss is:  0.1594734489917755
At  7448 /100 epoch loss is:  0.1762930303812027
At  7449 /100 epoch loss is:  0.17502601444721222
At  74

At  7599 /100 epoch loss is:  0.19350340962409973
At  7600 /100 epoch loss is:  0.1641618311405182
At  7601 /100 epoch loss is:  0.16986748576164246
At  7602 /100 epoch loss is:  0.1873909831047058
At  7603 /100 epoch loss is:  0.18650534749031067
At  7604 /100 epoch loss is:  0.16672764718532562
At  7605 /100 epoch loss is:  0.15211500227451324
At  7606 /100 epoch loss is:  0.16813211143016815
At  7607 /100 epoch loss is:  0.1567629873752594
At  7608 /100 epoch loss is:  0.16734318435192108
At  7609 /100 epoch loss is:  0.1747196912765503
At  7610 /100 epoch loss is:  0.15520983934402466
At  7611 /100 epoch loss is:  0.16862106323242188
At  7612 /100 epoch loss is:  0.16671106219291687
At  7613 /100 epoch loss is:  0.18590648472309113
At  7614 /100 epoch loss is:  0.1708381474018097
At  7615 /100 epoch loss is:  0.1714821308851242
At  7616 /100 epoch loss is:  0.1681767851114273
At  7617 /100 epoch loss is:  0.15774263441562653
At  7618 /100 epoch loss is:  0.1723012626171112
At  7619

At  7771 /100 epoch loss is:  0.1832219958305359
At  7772 /100 epoch loss is:  0.20615041255950928
At  7773 /100 epoch loss is:  0.19738346338272095
At  7774 /100 epoch loss is:  0.1942359209060669
At  7775 /100 epoch loss is:  0.18826499581336975
At  7776 /100 epoch loss is:  0.17438451945781708
At  7777 /100 epoch loss is:  0.17171339690685272
At  7778 /100 epoch loss is:  0.19012124836444855
At  7779 /100 epoch loss is:  0.23477691411972046
At  7780 /100 epoch loss is:  0.2565145790576935
At  7781 /100 epoch loss is:  0.24230565130710602
At  7782 /100 epoch loss is:  0.21092933416366577
At  7783 /100 epoch loss is:  0.21124255657196045
At  7784 /100 epoch loss is:  0.17874592542648315
At  7785 /100 epoch loss is:  0.21289631724357605
At  7786 /100 epoch loss is:  0.19265717267990112
At  7787 /100 epoch loss is:  0.16820241510868073
At  7788 /100 epoch loss is:  0.21161819994449615
At  7789 /100 epoch loss is:  0.20017920434474945
At  7790 /100 epoch loss is:  0.19831472635269165
At 

At  7936 /100 epoch loss is:  0.16771423816680908
At  7937 /100 epoch loss is:  0.1804324984550476
At  7938 /100 epoch loss is:  0.173280730843544
At  7939 /100 epoch loss is:  0.18432298302650452
At  7940 /100 epoch loss is:  0.1970500648021698
At  7941 /100 epoch loss is:  0.1685236692428589
At  7942 /100 epoch loss is:  0.17490635812282562
At  7943 /100 epoch loss is:  0.18818219006061554
At  7944 /100 epoch loss is:  0.19254007935523987
At  7945 /100 epoch loss is:  0.18031463027000427
At  7946 /100 epoch loss is:  0.17722472548484802
At  7947 /100 epoch loss is:  0.21993349492549896
At  7948 /100 epoch loss is:  0.19667205214500427
At  7949 /100 epoch loss is:  0.1848292350769043
At  7950 /100 epoch loss is:  0.18563401699066162
At  7951 /100 epoch loss is:  0.20390154421329498
At  7952 /100 epoch loss is:  0.18778781592845917
At  7953 /100 epoch loss is:  0.1873541623353958
At  7954 /100 epoch loss is:  0.17762188613414764
At  7955 /100 epoch loss is:  0.18600493669509888
At  795

At  8105 /100 epoch loss is:  0.16997383534908295
At  8106 /100 epoch loss is:  0.19494573771953583
At  8107 /100 epoch loss is:  0.19705843925476074
At  8108 /100 epoch loss is:  0.17422088980674744
At  8109 /100 epoch loss is:  0.18863344192504883
At  8110 /100 epoch loss is:  0.17575602233409882
At  8111 /100 epoch loss is:  0.17752523720264435
At  8112 /100 epoch loss is:  0.16733966767787933
At  8113 /100 epoch loss is:  0.1692284345626831
At  8114 /100 epoch loss is:  0.16498781740665436
At  8115 /100 epoch loss is:  0.18081700801849365
At  8116 /100 epoch loss is:  0.17850230634212494
At  8117 /100 epoch loss is:  0.177250936627388
At  8118 /100 epoch loss is:  0.16337276995182037
At  8119 /100 epoch loss is:  0.1762646585702896
At  8120 /100 epoch loss is:  0.16444139182567596
At  8121 /100 epoch loss is:  0.1759294867515564
At  8122 /100 epoch loss is:  0.16367796063423157
At  8123 /100 epoch loss is:  0.16222546994686127
At  8124 /100 epoch loss is:  0.16422756016254425
At  8

At  8272 /100 epoch loss is:  0.15615850687026978
At  8273 /100 epoch loss is:  0.191090390086174
At  8274 /100 epoch loss is:  0.18189096450805664
At  8275 /100 epoch loss is:  0.15300747752189636
At  8276 /100 epoch loss is:  0.17454968392848969
At  8277 /100 epoch loss is:  0.19698332250118256
At  8278 /100 epoch loss is:  0.1669878363609314
At  8279 /100 epoch loss is:  0.16418711841106415
At  8280 /100 epoch loss is:  0.1847575604915619
At  8281 /100 epoch loss is:  0.19457492232322693
At  8282 /100 epoch loss is:  0.17471928894519806
At  8283 /100 epoch loss is:  0.16509658098220825
At  8284 /100 epoch loss is:  0.18039506673812866
At  8285 /100 epoch loss is:  0.1702164113521576
At  8286 /100 epoch loss is:  0.18533876538276672
At  8287 /100 epoch loss is:  0.18644529581069946
At  8288 /100 epoch loss is:  0.1763840913772583
At  8289 /100 epoch loss is:  0.16062822937965393
At  8290 /100 epoch loss is:  0.1774260401725769
At  8291 /100 epoch loss is:  0.18174637854099274
At  829

At  8442 /100 epoch loss is:  0.18197426199913025
At  8443 /100 epoch loss is:  0.20791736245155334
At  8444 /100 epoch loss is:  0.22187411785125732
At  8445 /100 epoch loss is:  0.22093600034713745
At  8446 /100 epoch loss is:  0.1852678507566452
At  8447 /100 epoch loss is:  0.17701750993728638
At  8448 /100 epoch loss is:  0.2193778157234192
At  8449 /100 epoch loss is:  0.22077403962612152
At  8450 /100 epoch loss is:  0.18813368678092957
At  8451 /100 epoch loss is:  0.16193723678588867
At  8452 /100 epoch loss is:  0.18622863292694092
At  8453 /100 epoch loss is:  0.18715661764144897
At  8454 /100 epoch loss is:  0.17756715416908264
At  8455 /100 epoch loss is:  0.18129102885723114
At  8456 /100 epoch loss is:  0.1771877408027649
At  8457 /100 epoch loss is:  0.20577241480350494
At  8458 /100 epoch loss is:  0.1909334361553192
At  8459 /100 epoch loss is:  0.17724470794200897
At  8460 /100 epoch loss is:  0.1678711175918579
At  8461 /100 epoch loss is:  0.17581000924110413
At  8

At  8613 /100 epoch loss is:  0.17354144155979156
At  8614 /100 epoch loss is:  0.17216803133487701
At  8615 /100 epoch loss is:  0.16426943242549896
At  8616 /100 epoch loss is:  0.156063050031662
At  8617 /100 epoch loss is:  0.1689680814743042
At  8618 /100 epoch loss is:  0.16076886653900146
At  8619 /100 epoch loss is:  0.17312847077846527
At  8620 /100 epoch loss is:  0.15539374947547913
At  8621 /100 epoch loss is:  0.1616063416004181
At  8622 /100 epoch loss is:  0.16852906346321106
At  8623 /100 epoch loss is:  0.15424714982509613
At  8624 /100 epoch loss is:  0.16561545431613922
At  8625 /100 epoch loss is:  0.18550807237625122
At  8626 /100 epoch loss is:  0.18481536209583282
At  8627 /100 epoch loss is:  0.2132130116224289
At  8628 /100 epoch loss is:  0.15395177900791168
At  8629 /100 epoch loss is:  0.17456591129302979
At  8630 /100 epoch loss is:  0.18832704424858093
At  8631 /100 epoch loss is:  0.19618673622608185
At  8632 /100 epoch loss is:  0.17090603709220886
At  8

At  8794 /100 epoch loss is:  0.15021279454231262
At  8795 /100 epoch loss is:  0.1602686047554016
At  8796 /100 epoch loss is:  0.16229824721813202
At  8797 /100 epoch loss is:  0.15654738247394562
At  8798 /100 epoch loss is:  0.1498771756887436
At  8799 /100 epoch loss is:  0.1668676733970642
At  8800 /100 epoch loss is:  0.15432383120059967
At  8801 /100 epoch loss is:  0.15721826255321503
At  8802 /100 epoch loss is:  0.15995870530605316
At  8803 /100 epoch loss is:  0.1706331968307495
At  8804 /100 epoch loss is:  0.15651586651802063
At  8805 /100 epoch loss is:  0.16609132289886475
At  8806 /100 epoch loss is:  0.1650002896785736
At  8807 /100 epoch loss is:  0.15794093906879425
At  8808 /100 epoch loss is:  0.16295944154262543
At  8809 /100 epoch loss is:  0.15834054350852966
At  8810 /100 epoch loss is:  0.14405466616153717
At  8811 /100 epoch loss is:  0.17685136198997498
At  8812 /100 epoch loss is:  0.16800963878631592
At  8813 /100 epoch loss is:  0.15919332206249237
At  8

At  8964 /100 epoch loss is:  0.18404974043369293
At  8965 /100 epoch loss is:  0.1713452786207199
At  8966 /100 epoch loss is:  0.18132932484149933
At  8967 /100 epoch loss is:  0.18757253885269165
At  8968 /100 epoch loss is:  0.17557792365550995
At  8969 /100 epoch loss is:  0.17773588001728058
At  8970 /100 epoch loss is:  0.16306369006633759
At  8971 /100 epoch loss is:  0.1678246110677719
At  8972 /100 epoch loss is:  0.18750926852226257
At  8973 /100 epoch loss is:  0.18048004806041718
At  8974 /100 epoch loss is:  0.17387355864048004
At  8975 /100 epoch loss is:  0.18029525876045227
At  8976 /100 epoch loss is:  0.1621626615524292
At  8977 /100 epoch loss is:  0.1809125691652298
At  8978 /100 epoch loss is:  0.17550984025001526
At  8979 /100 epoch loss is:  0.18560469150543213
At  8980 /100 epoch loss is:  0.1775522530078888
At  8981 /100 epoch loss is:  0.18158915638923645
At  8982 /100 epoch loss is:  0.18048341572284698
At  8983 /100 epoch loss is:  0.20206882059574127
At  8

At  9133 /100 epoch loss is:  0.2304254174232483
At  9134 /100 epoch loss is:  0.20794059336185455
At  9135 /100 epoch loss is:  0.16690218448638916
At  9136 /100 epoch loss is:  0.2027653008699417
At  9137 /100 epoch loss is:  0.20607806742191315
At  9138 /100 epoch loss is:  0.15816722810268402
At  9139 /100 epoch loss is:  0.1843167245388031
At  9140 /100 epoch loss is:  0.15436652302742004
At  9141 /100 epoch loss is:  0.17291095852851868
At  9142 /100 epoch loss is:  0.16060195863246918
At  9143 /100 epoch loss is:  0.16444282233715057
At  9144 /100 epoch loss is:  0.1844741553068161
At  9145 /100 epoch loss is:  0.16880524158477783
At  9146 /100 epoch loss is:  0.17437797784805298
At  9147 /100 epoch loss is:  0.15081773698329926
At  9148 /100 epoch loss is:  0.15634693205356598
At  9149 /100 epoch loss is:  0.15212182700634003
At  9150 /100 epoch loss is:  0.15751178562641144
At  9151 /100 epoch loss is:  0.15309248864650726
At  9152 /100 epoch loss is:  0.16784760355949402
At  

At  9299 /100 epoch loss is:  0.1831624060869217
At  9300 /100 epoch loss is:  0.15821310877799988
At  9301 /100 epoch loss is:  0.1551615297794342
At  9302 /100 epoch loss is:  0.1655275970697403
At  9303 /100 epoch loss is:  0.1766299605369568
At  9304 /100 epoch loss is:  0.15544474124908447
At  9305 /100 epoch loss is:  0.17049390077590942
At  9306 /100 epoch loss is:  0.15023131668567657
At  9307 /100 epoch loss is:  0.15310482680797577
At  9308 /100 epoch loss is:  0.1715744584798813
At  9309 /100 epoch loss is:  0.1609160453081131
At  9310 /100 epoch loss is:  0.1496427357196808
At  9311 /100 epoch loss is:  0.17942163348197937
At  9312 /100 epoch loss is:  0.15369237959384918
At  9313 /100 epoch loss is:  0.1739722043275833
At  9314 /100 epoch loss is:  0.16473416984081268
At  9315 /100 epoch loss is:  0.16915905475616455
At  9316 /100 epoch loss is:  0.17522010207176208
At  9317 /100 epoch loss is:  0.17196528613567352
At  9318 /100 epoch loss is:  0.16807520389556885
At  9319

At  9469 /100 epoch loss is:  0.16876305639743805
At  9470 /100 epoch loss is:  0.18465805053710938
At  9471 /100 epoch loss is:  0.16242556273937225
At  9472 /100 epoch loss is:  0.17733508348464966
At  9473 /100 epoch loss is:  0.15979130566120148
At  9474 /100 epoch loss is:  0.1670113354921341
At  9475 /100 epoch loss is:  0.17135126888751984
At  9476 /100 epoch loss is:  0.1498507559299469
At  9477 /100 epoch loss is:  0.15003634989261627
At  9478 /100 epoch loss is:  0.187561497092247
At  9479 /100 epoch loss is:  0.15976303815841675
At  9480 /100 epoch loss is:  0.15662483870983124
At  9481 /100 epoch loss is:  0.15610335767269135
At  9482 /100 epoch loss is:  0.17532490193843842
At  9483 /100 epoch loss is:  0.16724233329296112
At  9484 /100 epoch loss is:  0.1552560180425644
At  9485 /100 epoch loss is:  0.16540555655956268
At  9486 /100 epoch loss is:  0.18932858109474182
At  9487 /100 epoch loss is:  0.2058641016483307
At  9488 /100 epoch loss is:  0.18496577441692352
At  94

At  9642 /100 epoch loss is:  0.16495749354362488
At  9643 /100 epoch loss is:  0.15786761045455933
At  9644 /100 epoch loss is:  0.16914565861225128
At  9645 /100 epoch loss is:  0.16151779890060425
At  9646 /100 epoch loss is:  0.16005750000476837
At  9647 /100 epoch loss is:  0.16962429881095886
At  9648 /100 epoch loss is:  0.1717512160539627
At  9649 /100 epoch loss is:  0.1589985340833664
At  9650 /100 epoch loss is:  0.14963799715042114
At  9651 /100 epoch loss is:  0.16813242435455322
At  9652 /100 epoch loss is:  0.17416125535964966
At  9653 /100 epoch loss is:  0.18210318684577942
At  9654 /100 epoch loss is:  0.17948393523693085
At  9655 /100 epoch loss is:  0.16999690234661102
At  9656 /100 epoch loss is:  0.16566520929336548
At  9657 /100 epoch loss is:  0.170747309923172
At  9658 /100 epoch loss is:  0.14821526408195496
At  9659 /100 epoch loss is:  0.17477349936962128
At  9660 /100 epoch loss is:  0.15807299315929413
At  9661 /100 epoch loss is:  0.16106866300106049
At  

At  9810 /100 epoch loss is:  0.18224453926086426
At  9811 /100 epoch loss is:  0.20685885846614838
At  9812 /100 epoch loss is:  0.17999885976314545
At  9813 /100 epoch loss is:  0.1711704134941101
At  9814 /100 epoch loss is:  0.17092300951480865
At  9815 /100 epoch loss is:  0.18925927579402924
At  9816 /100 epoch loss is:  0.18054305016994476
At  9817 /100 epoch loss is:  0.15719255805015564
At  9818 /100 epoch loss is:  0.1747794896364212
At  9819 /100 epoch loss is:  0.178758442401886
At  9820 /100 epoch loss is:  0.16173143684864044
At  9821 /100 epoch loss is:  0.18388481438159943
At  9822 /100 epoch loss is:  0.18582651019096375
At  9823 /100 epoch loss is:  0.15699972212314606
At  9824 /100 epoch loss is:  0.17846445739269257
At  9825 /100 epoch loss is:  0.17726005613803864
At  9826 /100 epoch loss is:  0.16275684535503387
At  9827 /100 epoch loss is:  0.16876474022865295
At  9828 /100 epoch loss is:  0.18391497433185577
At  9829 /100 epoch loss is:  0.15959154069423676
At  

At  9992 /100 epoch loss is:  0.1724957823753357
At  9993 /100 epoch loss is:  0.15121500194072723
At  9994 /100 epoch loss is:  0.15552622079849243
At  9995 /100 epoch loss is:  0.1564360111951828
At  9996 /100 epoch loss is:  0.1657053828239441
At  9997 /100 epoch loss is:  0.1463632434606552
At  9998 /100 epoch loss is:  0.1691063493490219
At  9999 /100 epoch loss is:  0.15422864258289337


In [36]:
output_check = (model(train_data))
output_check

tensor([[ -18.2097,    9.2705,   -0.3725,  -27.2269],
        [ -68.0196,  -17.2184,   -1.2256,   -3.7805],
        [ -55.5025,  -10.6432,   -0.2057,   -9.5235],
        ...,
        [-164.9006,  -50.5300,   -0.2529,   14.8606],
        [  34.3166,   27.5726,    1.1588,  -31.8794],
        [-122.6070,  -36.2014,   -2.8881,    0.7798]], grad_fn=<AddmmBackward>)

In [38]:
def accuracy(output_check,train_label):
    list1 = []
    for e in output_check:
        max_index = 0
        for i in range(len(e)):
            if e[i].item() > e[max_index].item():
                max_index = i
        list1.append(max_index)

    count = 0
    for x,y in zip(list1, train_label):
        if x!=y:
            count+=1
    return (len(list1)-count)/len(list1)

print('Accuracy: ', accuracy(output_check, train_label)*100)

Accuracy:  94.0
